clash_lib/src/proxy/tuic/mod.rs

```diff
@@ -144,34 +143,46 @@ impl Handler {
         crypto.enable_early_data = true;
         crypto.enable_sni = !opts.disable_sni;
         let mut quinn_config = QuinnConfig::new(Arc::new(crypto));
-        let mut quinn_transport_config = QuinnTransportConfig::default();
-        quinn_transport_config
+        let mut transport_config = QuinnTransportConfig::default();
+        transport_config
             .max_concurrent_bidi_streams(opts.max_open_stream)
             .max_concurrent_uni_streams(opts.max_open_stream)
             .send_window(opts.send_window)
             .stream_receive_window(opts.receive_window)
-            .max_idle_timeout(None)
-            .congestion_controller_factory(Arc::new(CubicConfig::default()));
-        quinn_config.transport_config(Arc::new(quinn_transport_config));
+            .max_idle_timeout(Some(opts.idle_timeout.into()));
+        match opts.congestion_controller {
+            CongestionControl::Cubic => {
+                transport_config.congestion_controller_factory(Arc::new(CubicConfig::default()))
+            }
+            CongestionControl::NewReno => {
+                transport_config.congestion_controller_factory(Arc::new(NewRenoConfig::default()))
+            }
+            CongestionControl::Bbr => {
+                transport_config.congestion_controller_factory(Arc::new(BbrConfig::default()))
+            }
+        };
+
+        quinn_config.transport_config(Arc::new(transport_config));
         // Try to create an IPv4 socket as the placeholder first, if it fails, try IPv6.
+        // If it dont match server's ipv4/ipv6, rebind is needed
         let socket =
             UdpSocket::bind(SocketAddr::from((Ipv4Addr::UNSPECIFIED, 0))).or_else(|err| {
```

Comment:
i have a sense that the iOS issue in the group is somehow related to this bind iirc iOS doesn't allow bind to 0.0.0.0. I will take a look again and might change here